src/Diagnostics.RuntimeHost/Controllers/Site/SitesController.cs

```diff
@@ -116,15 +182,65 @@ public async Task<IActionResult> GetInsights(string subscriptionId, string resou
             return await base.GetInsights(app, pesId, supportTopicId, startTime, endTime, timeGrain);
         }
 
+        /// <summary>
+        /// Publish detector package.
+        /// </summary>
+        /// <param name="pkg">The package.</param>
+        /// <returns>Task for publishing package.</returns>
         [HttpPost(UriElements.Publish)]
-        public async Task<IActionResult> PublishDetector(string subscriptionId, string resourceGroupName, string siteName, [FromBody] DetectorPackage pkg)
```

Comment:
>string subscriptionId, string resourceGroupName, string siteName,  [](start = 57, length = 66)

You will need this as the URI path requires it

Reply:
Discussed offline. Keep the change.